src/popup/components/TransactionTagList.vue

```diff
@@ -63,77 +70,88 @@ export default defineComponent({
 
     const labels = computed<(string | TranslateResult)[]>(() => {
       if (!props.tx) return [];
+      const externalLabels = [];
+      let innerLabels = [];
+
+      if (getTxType.value(props.tx) === TX_TYPE_MDW.GAMetaTx) {
+        externalLabels.push(i18n.t('transaction.type.gaMetaTx'));
+      }
+      if (getTxType.value(props.tx) === TX_TYPE_MDW.PayingForTx) {
+        externalLabels.push(i18n.t('transaction.type.payingForTx'));
+      }
```

Comment:
This computed property is already really messed up. Adding additional logic and two separate arrays of labels makes it almost unreadable. But first of all no need to use additional `getTxType`. You can use the `txType.value` that already exists. And then just build one array of labels with a list of ifs.

Reply:
IDEM

Reply:
OK, so what I suggest is to use the `useTransactionTx` twice: for the `props.tx` and for the innerTx. Or extend this composable to return itself again for the inner tx:
```
// This goes inside of the useTransactionTx
// And there is a naming collision because `innerTx` is already taken, so it should be replaced with just `tx`.
const innerTx = computed(() => useTransactionTx({
  store,
  tx: <the_check_if_the_tx_has_innerTx> ? tx.value.tx?.tx : undefined,
}));
...
return {
  ...
  innerTx,
}
```

this way here we would have:
```
const {
  txType,
  isAllowance,
  isDex,
  innerTx,
} = useTransactionTx({ store: root.$store, tx });

const {
  txType: innerTxType,
} = innerTx.value
```

Reply:
Well, ofc we can do that. But all the info i want to get from `useTransactionTx` composable related only to an innerTx, not an external one.

Reply:
Sure, if you think my propositions are bad leave it as it is. I just don't like how this logic is becoming an unreadable behemoth. Previously it was rather bad and now it's even worse.

Reply:
Another idea: maybe the `transactionTx` composable could return computed property `innerTxType`?